public/pages/Dashboard/Components/AnomaliesDistribution.tsx

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+import { Detector } from '../../../models/interfaces';
+import { getDetectorResults } from '../../../redux/reducers/anomalyResults';
+import { useState, useEffect } from 'react';
+import { buildGetRecentAnomalyResultQuery, fillOutColors } from '../utils';
+import ContentPanel from '../../../components/ContentPanel/ContentPanel';
+import { EuiSelect, EuiText, EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import { Chart, Partition } from '@elastic/charts';
+import { useDispatch } from 'react-redux';
+import { AD_DOC_FIELDS } from '../../../../server/utils/constants';
+import { Datum } from '@elastic/charts/dist/utils/commons';
+import React from 'react';
+
+export interface AnomaliesDistributionChartProps {
+  allDetectorsSelected: boolean;
+  selectedDetectors: Detector[];
+}
+
+export const AnomaliesDistributionChart = (
+  props: AnomaliesDistributionChartProps
+) => {
+  const dispatch = useDispatch();
+
+  const [anomalyResult, setAnomalyResult] = useState([] as object[]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [indicesNumber, setIndicesNumber] = useState(0);
+
+  const timeRangeOptions = [
+    { value: '24h', text: 'Last 24 hours' },
+    { value: '7d', text: 'Last 7 days' },
+  ];
+  const [timeRange, setTimeRange] = useState(timeRangeOptions[0].value);
+
+  const getRecentAnomalyResultQuery = buildGetRecentAnomalyResultQuery(
+    timeRange
+  );
+
+  const getAnomalyResult = async (currentDetectors: Detector[]) => {
+    const currentAnomalyResult = [] as object[];
+    const indicesSet = new Set();
+    for (let detector of currentDetectors) {
+      const resp = await dispatch(
+        getDetectorResults(detector.id, getRecentAnomalyResultQuery)
+      );
+      currentAnomalyResult.push({
+        detector: detector,
+        results: resp.data.response.results,
+      });
+      indicesSet.add(detector.indices.toString());
+    }
+    setAnomalyResult(currentAnomalyResult);
+    setIndicesNumber(indicesSet.size);
+    setIsLoading(false);
```

Comment:
The problem is the that on the Dashboard page, if there are 2 or more components using same redux, I will encounter the following issues:

Issue 1: infinite refreshing. Dashboard.tsx and AnomaliesLiveChart.tsx is in the same Dashboard page. If both Dashboard.tsx and AnomaliesLiveChart.tsx uses `const isLoading = useSelector((state: AppState) => state.ad.requesting);` and it cause infinite refreshing of the page because both components use that `ad` redux to get data, but `ad.requesting` keeps changing, and Dashboard page keeps refreshing.

Issue 2: always refresh both components AnomaliesLiveChart.tsx and AnomaliesDistribution.tsx, while only AnomaliesLiveChart needs refreshing every 1min, if both of them use  `const isLoading = useSelector((state: AppState) => state.anomalyResults.requesting);`. My understanding is that this is caused by auto refresh of AnomaliesLiveChart, and `state.anomalyResults.requesting` is shared by those 2 components. Once `requesting` changes to true by AnomaliesLiveChart, the UI of AnomaliesDistribution is changed as well, but AnomaliesDistribution does not make API call. 

Those are the reasons I avoid using the global redux. And code here is my workaround. Please let me know your thoughts on how to use such redux appropriately.

Reply:
I'll need to rethink this but to unblock, can we name isLoading to appropriate to relate better such as `setAnomalyResultsLoading`

Reply:
> I'll need to rethink this but to unblock, can we name isLoading to appropriate to relate better such as `setAnomalyResultsLoading`
Sure. Will rename it.
> Few question:
> 
> * Shouldn't both charts refresh together always ? (i.e let's if we want to make some live charts, it has to be in sync and on same time. IMO.
I checked with UX, she confirmed that only Live chart should be refreshed every min, while this distribution sunburst chart is static unless user chooses new detectors